### PR TITLE
Refactor how `TermInfos` are handled in the `dumpProcessor`

### PR DESCRIPTION
**Pull request by hankewyczz**
# Purpose

_In 2-3 sentences - what does this code actually do, and why?_

Our workflow with regards to `TermInfos` was poorly designed — you'd need two scrapes to populate them in the DB! (spoiler - that was my fault)

Imagine a new term, with ID `3`, was just released

Before, the flow was:
- Each time we run a scrape, the `TermInfo` table was updated 
  - Only the `TermInfo`s with class/section data in the database would be kept
  - However - this was done BEFORE the sections/classes just scraped were added
  - So - at the time of the check, there are no sections/classes with term ID `3`!
  - The class/section data for `3` would be inserted.

Then, the next time we ran a scrape, term `3` would be added. 

This takes care of that (finally).

# Tickets

_What Trello tickets (if any) are associated with this PR?_

- N/A

# Reviewers

Primary reviewer:

- Pick one primary reviewer
  - The team member with the most relevant knowledge about the code area this PR touches
  - **NOT** an author of the PR
  - If the primary reviewer is the project lead, _select two primary reviewers_
    - Goal: facilitate knowledge transfer to other team members
  - Primary reviewers **are required to approve the PR** before it can be merged

**Primary**:

Use the **"Reviewers"** feature in Github

Secondary reviewers:

- Pick as many as appropriate — anyone who would benefit from reading the PR
  - Tag them using their Github usernames below

**Secondary**:

### CODE REVIEW

File: scrapers/classes/parsersxe/bannerv9Parser.ts [hunk 1]
```diff
@@ -98,25 +98,6 @@ export class Bannerv9Parser {
     });
   }
 
-  /**
```
Comment (hankewyczz) [inline scrapers/classes/parsersxe/bannerv9Parser.ts:101]: This doesn't really concern Banner, so I moved it out of here

File: services/dumpProcessor.ts [hunk 2]
```diff
@@ -46,9 +46,12 @@ class DumpProcessor {
 
     await this.saveSubjectsToDatabase(termDump.subjects);
 
-    await this.saveTermInfosToDatabase(currentTermInfos);
+    await this.saveTermInfosToDatabase(allTermInfos);
 
-    if (destroy) {
+    if (deleteOutdatedData) {
```
Comment (hankewyczz) [inline services/dumpProcessor.ts:51]: Updated the naming to make it clearer and added some comments. Not directly related to the PR, but a minor tweak

**Review by Lucas-Dunker — CHANGES_REQUESTED** [review]
Great PR, well explained AND with tests 😎 Only revisions I can think of are to fix the linting warnings!

**Comment by hankewyczz** [comment]
@Lucas-Dunker Generally, for lint warnings - if your changes caused the lint warning, you're responsible. If not, it's probably left for a different PR. In this case, those lint warnings were already there and don't have anything to do with this PR, so we want to leave it for later - don't want PRs to be too cluttered

**Review by Lucas-Dunker — APPROVED** [review]
LGTM 👍

**Review by pranavphadke1 — APPROVED** [review]

**Pull request merged**